shared/src/simulation/behaviors/assign-leader.ts

```diff
@@ -83,29 +84,24 @@ export const assignLeaderBehavior: SimulationBehavior<AssignLeaderBehaviorState>
                 case 'tickEvent':
                     {
                         if (!behaviorState.leaderId) {
-                            const personnel = Object.values(
-                                draftState.personnel
-                            ).filter(
-                                (pers) =>
-                                    isInSpecificSimulatedRegion(
-                                        pers,
-                                        simulatedRegion.id
-                                    ) && pers.personnelType !== 'notarzt'
+                            selectNewLeader(
+                                draftState,
+                                simulatedRegion,
+                                behaviorState
                             );
+                        }
+                    }
+                    break;
+                case 'personnelRemovedEvent':
+                    {
+                        // if the leader is removed from the region the value is unset and a new leader is selected
 
-                            if (personnel.length === 0) {
-                                return;
-                            }
-
-                            personnel.sort(
-                                (a, b) =>
-                                    personnelPriorities[b.personnelType] -
-                                    personnelPriorities[a.personnelType]
-                            );
-                            changeLeader(
+                        if (event.personnelId === behaviorState.leaderId) {
+                            behaviorState.leaderId = undefined;
```

Comment:
```suggestion
```
This would create an event with wrong previous leader.

Reply:
i changed that portion of the code